app/controllers/items_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ItemsController < ApplicationController
+  def index
+    if params[:merchant_id].nil?
+      @items = Item.all
+    else
+      @merchant = Merchant.find(params[:merchant_id])
+
+      @items = @merchant.items
+    end
+  end
+
+  def show
+    @item = Item.find(params[:id])
+    @merchant = Merchant.find(@item.merchant_id)
+  end
+
+  def new
+    @merchant = Merchant.find(params[:merchant_id])
+  end
+
+  def create
+    binding.pry
+  end
+
+  def edit
+    @item = Item.find(params[:id])
+  end
+
+  def update
+    @item = Item.find(params[:id])
+    Item.update(item_params)
+    redirect_to "/items/#{@item.id}"
```

Comment:
since this action is redirecting to a view instead of rendering one, you don't need to use an instance variable for item since we are not trying to use that item to display on a certain view.